### ISSUE TITLE
Adjust colors based on window focus, like native apps

### ISSUE THREAD

**Issue by GroovinChip** [macosui/macos_ui#395]
![Image](https://user-images.githubusercontent.com/4250470/226665084-cf21265a-8606-4b2f-9dad-f267b332f545.png)

This will involve either using the `desktop_lifecycle` plugin or writing our own.

I've attempted this with `desktop_lifecycle` and it works quite well.

We'll need to add a dynamic color to `MacosColors` to represent inactive colors for the sidebar and toolbar (both use the same), and a style in `MacosTypography` that colors inactive text in the sidebar and toolbar. We'll also need inactive icon colors, and a utility to darken the primary color where appropriate.

We will need to determine whether to check if the window is active in each widget that needs to be adjusted, or do it top-level (like in `MacosApp`) and propagate the result of the check through the widget tree via `MacosThemeData` or another `InheritedWidget`.

**Comment by Adrian-Samoticha** [comment]
[macos_window_utils](https://pub.dev/packages/macos_window_utils)’s `NSWindowDelegate` can also be used to detect window focus events and since we’ll already be depending on that, it might be a better idea to use that, instead (assuming cross-platform support isn’t targetted, of course).

**EDIT:** That said, macos_window_utils uses `NSWindowDelegate` while desktop_lifecycle uses `NotificationCenter`, so perhaps desktop_lifecycle might be a better choice compatibility-wise.

**EDIT 2:** Nevermind the first edit, we are already using `NSWindowDelegate` anyway because we need `NSAppPresentationOptions`, so the compatibility argument doesn’t actually hold.

**Comment by glanium** [comment]
and Please add widgets equivalent to material InkWell Widget.

InkWell widget provides focus visual change and mouse hover effects.
Macos UI's ListTile can't get focus.
Material's ListTile is built on based on InkWell then can get focus.

Thx.

**Comment by GroovinChip** [comment]
> and Please add widgets equivalent to material InkWell Widget.
> 
> 
> 
> InkWell widget provides focus visual change and mouse hover effects.
> 
> Macos UI's ListTile can't get focus.
> 
> Material's ListTile is built on based on InkWell then can get focus.
> 
> 
> 
> Thx.
> 
> 

Please file a new issue for this, as it is not connected to this.

**Comment by glanium** [comment]
Ok
Done. https://github.com/macosui/macos_ui/issues/399
Thx.

**Comment by ralph-bergmann** [comment]
It works with the [system's accent color](https://github.com/macosui/macos_ui/issues/486#issuecomment-1764197632):


<img width="944" alt="SCR-20231207-meta" src="https://github.com/macosui/macos_ui/assets/1191649/718c3eff-c9f0-47d9-a132-795d8abe4abc">
<img width="962" alt="SCR-20231207-mevn" src="https://github.com/macosui/macos_ui/assets/1191649/8aa58580-4fb4-49ac-8695-ba5bdaf17d26">

**Comment by Adrian-Samoticha** [comment]
> It works with the [system's accent color](https://github.com/macosui/macos_ui/issues/486#issuecomment-1764197632):
> 



It works with widgets that have the accent color already implemented. Currently, that only includes the buttons and the sidebar items, once #484 is merged.
